src/cargo-fmt/main.rs

```diff
@@ -112,20 +112,13 @@ fn execute() -> i32 {
 
     let strategy = CargoFmtStrategy::from_opts(&opts);
     let mut rustfmt_args = opts.rustfmt_options;
-    if opts.check {
-        let check_flag = String::from("--check");
-        if !rustfmt_args.contains(&check_flag) {
-            rustfmt_args.push(check_flag);
-        }
+    let cargo_fmt_check = opts.check;
+    if let Err(ref msg) =
+        build_rustfmt_args(opts.message_format, &mut rustfmt_args, cargo_fmt_check)
```

Comment:
Maybe you can pass the getopts `Options` as reference, this way `build_rustfmt_args` would access whatever it needs ?

```rust
        build_rustfmt_args(&opts, &mut rustfmt_args)
```

Reply:
SGTM